### DISCUSSION TITLE
sql dialect column type comparison table?

### DISCUSSION THREAD

**Discussion by wu-clan** [sqlalchemy/sqlalchemy#11907] [Usage Questions] [answered]
I would like to know if there exists a column type cross-reference table for different dialects, which would be very helpful for us to implement the type conversion.

**Comment by CaselIT** [comment]
Hi,

I don't understand what you mean by `column type cross-reference table`.
You mean something like what `Text` will be rendered to in oracle, postgresql, mysql, etc for the various types used by sqlalchemy?

**Reply by wu-clan** [reply]
Yes, I think you're right

Sorry, my English skills are weak. 😅

**Reply by CaselIT** [reply, marked as answer]
No there isn't any table, but it shouldn't be hard to find out:
```py
import sqlalchemy as sa
from sqlalchemy.dialects import postgresql, oracle

for d in (postgresql.dialect(), oracle.dialect()):
    print(d.name)
    for type in [sa.Text, sa.Integer]:
        print(type.__name__, ':', d.type_compiler.process(type()))

**Reply by wu-clan** [reply]
Thanks. Great job.
